### PR TITLE
add test suite and microservice document

### PR DESCRIPTION
**Pull request by vivian-zhou-skydio**
Added microservice document with proofreading changes. Also, added Amy's test suite into designated folder and updated the README.md in the microservices folder.

Previous PR: https://github.com/Skydio/aircam/pull/51975

### CODE REVIEW

File: en/services/illuminator.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+Along with this, the illuminator protocol also publishes status information for developers or users. The status messaging encompasses the current configuration of the illuminator and the health of the illuminator device.
+   
+
+## Concepts
+
+### MAVLink Illuminator Implementations
+
+These illuminators have built-in MAVLink support:
+- Skydio Spotlight
```
Comment (jenke-skydio) [inline en/services/illuminator.md]: align on marketing name here.  not sure if its `Skydio Spotlight` `Skydio Spotlight for X10` `Skydio X10 Spotlight` or something else

**Review by jenke-skydio — APPROVED** [review]
looks good.  Please check on the marketing name

File: en/services/illuminator.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+### Message/Enum Summary
+| Message                                                                                 | Description                                                                                                                         |
+| --------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------- |
+| <span id="COMPONENT_INFORMATION_BASIC"></span>[COMPONENT_INFORMATION_BASIC](../messages/common.md#COMPONENT_INFORMATION_BASIC) | Basic illuminator information data. Should be requested using MAV_CMD_REQUEST_MESSAGE on startup, or when required.|
```
Comment (amy-chen-skydio) [inline en/services/illuminator.md]: link `MAV_CMD_REQUEST_MESSAGE`?

File: en/services/illuminator.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+| Enum                                                                                             | Description               |
+| ------------------------------------------------------------------------------------------------ | ------------------------- |
+| <span id="ILLUMINATOR_MODE"></span>[ILLUMINATOR_MODE](../messages/common.md#ILLUMINATOR_MODE)    | Illuminator modes. |
+| <span id="ILLUMINATOR_ERROR_FLAGS"></span>[ILLUMINATOR_ERROR_FLAGS](../messages/common.md#ILLUMINATOR_ERROR_FLAGS)    | Fault/health indications.             |
+
+
+### Illuminator Connection
+
+Illuminators are expected to follow the [Heartbeat/Connection Protocol](https://github.com/mavlink/mavlink-devguide/blob/master/en/services/heartbeat.md) and send a constant flow of heartbeats (nominally at 1Hz). Illuminators are distinguished via their unique component ID: [MAV_COMP_ID_ILLUMINATOR](#MAV_COMP_ID_ILLUMINATOR). Once a heartbeat is received, the drone can then send a REQUEST_MESSAGE command to the illuminator to receive information, set settings, or control the illuminator. An example below illustrates how a drone can request the status of the illuminator.
```
Comment (amy-chen-skydio) [inline en/services/illuminator.md]: `MAV_CMD_REQUEST_MESSAGE`

File: en/services/illuminator.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+
+## Implementation and Messages
+
+#### COMPONENT\_INFORMATION\_BASIC
+While the MAV\_TYPE and Component ID help identify the system and component, the [COMPONENT_INFORMATION_BASIC](#COMPONENT_INFORMATION_BASIC) command can be requested to retrieve component information data, which can help further identify the component being communicated with. This data includes time\_boot\_ms, MAV\_PROTOCOL\_CAPABILITY, vendor\_name, model\_name, software\_version, hardware\_version, and serial\_number.
```
Comment (amy-chen-skydio) [inline en/services/illuminator.md]: > time_boot_ms, MAV_PROTOCOL_CAPABILITY, vendor_name, model_name, software_version, hardware_version, and serial_number.

put these in `code` form or just reference the table?

File: en/services/illuminator.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+Illuminators can be set in different modes which can change the behavior of the illuminator (described in a separate section, [Modes](#MODES) section. The brightness can be set via "Brightness" as a percentage value (0-100%). Illuminators may also have the functionality to strobe the light source. This behavior is configured via "Strobe Period" and "Strobe Duty". These parameters can be set to 0 when not used. "Strobe Period" is in seconds and "Strobe Duty" is a percentage value (indicating the % of time in the "Strobe Period" the illuminator is enabled).
+<!-- This blank line provides visual separation in the document -->
+<!-- This blank line provides visual separation in the document -->
+##### MODES
+Illuminator modes can help the drone configure the interaction behavior of the illuminator. 
+
+If the mode is unknown, the mode parameter value will be set to 0. 
+
+The mode value as 1 indicates ILLUMINATOR\_MODE\_INTERNAL\_CONTROL where the illuminator behavior is controlled by [MAV_CMD_DO_ILLUMINATOR_CONFIGURE](#MAV_CMD_DO_ILLUMINATOR_CONFIGURE) settings.
```
Comment (amy-chen-skydio) [inline en/services/illuminator.md]: `ILLUMINATOR_MODE_INTERNAL_CONTROL` or link?

File: en/services/illuminator.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+<!-- This blank line provides visual separation in the document -->
+<!-- This blank line provides visual separation in the document -->
+##### MODES
+Illuminator modes can help the drone configure the interaction behavior of the illuminator. 
+
+If the mode is unknown, the mode parameter value will be set to 0. 
+
+The mode value as 1 indicates ILLUMINATOR\_MODE\_INTERNAL\_CONTROL where the illuminator behavior is controlled by [MAV_CMD_DO_ILLUMINATOR_CONFIGURE](#MAV_CMD_DO_ILLUMINATOR_CONFIGURE) settings. 
+
+When the value is set to 2, it indicates that the illuminator mode is ILLUMINATOR\_MODE\_EXTERNAL\_SYNC. This mode is for instances where the illuminator behavior is controlled by external factors: e.g. an external hardware signal.
```
Comment (amy-chen-skydio) [inline en/services/illuminator.md]: `ILLUMINATOR_MODE_EXTERNAL_SYNC` or link?

File: en/services/illuminator.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+<!-- This blank line provides visual separation in the document -->
+<!-- This blank line provides visual separation in the document -->
+##### STATUS
+The [ILLUMINATOR_STATUS](#ILLUMINATOR_STATUS) message can be requested to receive information about the status of the illuminator. This includes information such as uptime, errors, whether the illuminator is enabled via [MAV_CMD_ILLUMINATOR_ON_OFF](#MAV_CMD_ILLUMINATOR_ON_OFF), current settings from the [MAV_CMD_DO_ILLUMINATOR_CONFIGURE](#MAV_CMD_DO_ILLUMINATOR_CONFIGURE) command, and temperature of the illuminator. 
+<!-- This blank line provides visual separation in the document -->
+<!-- This blank line provides visual separation in the document -->
+##### ILLUMINATOR\_ERROR\_FLAGS
+The [ILLUMINATOR_ERROR_FLAGS](#ILLUMINATOR_ERROR_FLAGS) can be used to indicate if there is any issue with the illuminator. At this time, there are three flags. If there is no error and the illuminator is behaving as normal, the ILLUMINATOR\_ERROR\_FLAGS bitmap will be 0. If ILLUMINATOR\_ERROR\_FLAGS is set to 1 or 2, this indicates an error related to the temperature of the illuminator.
```
Comment (amy-chen-skydio) [inline en/services/illuminator.md]: replace ILLUMINATOR\_ERROR\_FLAGS with [ILLUMINATOR_ERROR_FLAGS](#ILLUMINATOR_ERROR_FLAGS)?

File: en/services/illuminator.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+<!-- This blank line provides visual separation in the document -->
+<!-- This blank line provides visual separation in the document -->
+##### ILLUMINATOR\_ERROR\_FLAGS
+The [ILLUMINATOR_ERROR_FLAGS](#ILLUMINATOR_ERROR_FLAGS) can be used to indicate if there is any issue with the illuminator. At this time, there are three flags. If there is no error and the illuminator is behaving as normal, the ILLUMINATOR\_ERROR\_FLAGS bitmap will be 0. If ILLUMINATOR\_ERROR\_FLAGS is set to 1 or 2, this indicates an error related to the temperature of the illuminator. 
+
+"1" indicates ILLUMINATOR\_ERROR\_FLAGS\_THERMAL\_THROTTLING as in the illuminator is throttling its output due to a thermal issue. "2" indicates ILLUMINATOR\_ERROR\_FLAGS\_OVER\_TEMPERATURE\_SHUTDOWN which means that the illuminator is shutting off due to passing some temperature threshold.
```
Comment (amy-chen-skydio) [inline en/services/illuminator.md]: `ILLUMINATOR_ERROR_FLAGS_THERMAL_THROTTLING` / `ILLUMINATOR_ERROR_FLAGS_OVER_TEMPERATURE_SHUTDOWN` or link?

**Pull request closed without merging**